### PR TITLE
Vite: Use mdx2 babel pre-processing

### PR DESCRIPTION
**Pull request by IanVS**
Issue:

## What I did

This incorporates https://github.com/storybookjs/mdx2-csf/pull/30 into the vite builder, removing the use of the vite react-plugin, freeing us to support vite 3 and 4!

Note: we'll want to update the version of `mdx2-csf` once the referenced PR is merged / released.  

## How to test

I tested this out by creating a react sandbox, and testing both build and dev.  HMR even seems to work in dev!

### CODE REVIEW

**Comment by ndelangen** [comment]
This is awesome!

File: code/lib/builder-vite/src/plugins/mdx-plugin.ts [hunk 1]
```diff
@@ -46,33 +30,16 @@ export function mdxPlugin(options: Options): Plugin {
         },
       });
 
-      const mdxCode = String(
+      const code = String(
         await compile(src, {
           skipCsf: !isStorybookMdx(id),
           ...mdxLoaderOptions,
```
Comment (ndelangen) [inline code/lib/builder-vite/src/plugins/mdx-plugin.ts:36]: You're missing `jsxOptions`

Comment (ndelangen) [reply code/lib/builder-vite/src/plugins/mdx-plugin.ts:36]: Yeah, I recon we want to get them from a similar place as mdxCompilerOptions: from a preset value

**Comment by shilman** [comment]
@IanVS @ndelangen this is great. merging this for now and @ndelangen you can take that in a separate PR

**Pull request merged**